test/test_torch.py

```diff
@@ -11888,6 +11888,229 @@ def test_fft_input_modification(self, device):
         _ = torch.irfft(half_spectrum_copy, 2, signal_sizes=(2, 2))
         self.assertEqual(half_spectrum, half_spectrum_copy)
 
+    @onlyOnCPUAndCUDA
+    @unittest.skipIf(not TEST_MKL, "PyTorch is built without MKL support")
+    @dtypes(torch.double)
+    def test_istft_round_trip_simple_cases(self, device, dtype):
+        """stft -> istft should recover the original signale"""
+        def _test(input, n_fft, length):
+            stft = torch.stft(input, n_fft=n_fft)
+            inverse = torch.istft(stft, n_fft=n_fft, length=length)
+            self.assertEqual(input, inverse, exact_dtype=True)
+
+        _test(torch.ones(4, dtype=dtype, device=device), 4, 4)
+        _test(torch.zeros(4, dtype=dtype, device=device), 4, 4)
+
+    @onlyOnCPUAndCUDA
+    @unittest.skipIf(not TEST_MKL, "PyTorch is built without MKL support")
+    @dtypes(torch.double)
+    def test_istft_round_trip_various_params(self, device, dtype):
+        """stft -> istft should recover the original signale"""
+        def _test_istft_is_inverse_of_stft(kwargs):
+            # generates a random sound signal for each tril and then does the stft/istft
+            # operation to check whether we can reconstruct signal
+            data_sizes = [(2, 20), (3, 15), (4, 10)]
+            num_trials = 100
+            for sizes in data_sizes:
+                for i in range(num_trials):
+                    original = torch.randn(*sizes, dtype=dtype, device=device)
+                    stft = torch.stft(original, **kwargs)
```

Comment:
> > * Batch packing/unpacking is performed in Python. ATen implementation expects 4D input tensor.
> 
> I'm assuming you did so to mirror the `stft` implementation? This means that the libtorch implementation will behave differently from the python implementation, right?

I picked this because it was much simpler to handle variable length array on Python side. I think it is overall simple to have `istft` focuses on `stft` output, which is either 3D or 4D. The current `aten:istft` implementation only handles 4D. So I would change that to handle 3D and 4D. Also Python/libtorch should be able to focus 3D/4D.  

> > * The way `hop_length` is initialized in the same way as `stft` implementation. [The Torchaudio's version tried to mimic the same behavior but slightly different](https://github.com/pytorch/audio/blob/7da61a4beeec7b7ff9ff5f1532b2adf99220a9b1/torchaudio/functional.py#L152-L157).
> 
> The initialization in ATen seems fine to me. Can you clarify if it changes something for the user in python?

* `aten::istft`

```
auto hop_length = hop_lengthOpt.value_or(n_fft >> 2);
auto win_length = win_lengthOpt.value_or(n_fft);
```

* `torchaudio.functional.istft`

```
if win_length is None:
    win_length = n_fft
if hop_length is None:
    hop_length = int(win_length // 4)
```

When `hop_length` and if `win_length` is given and the value is different than `n_fft`, then `hop_length` is initialized to a different values (`win_length//4` vs `n_fft//4`). The comment on `torchaudio` side (`# use stft defaults for Optionals`) suggests that the original author's intention but this case is not covered.

Reply:
> `stft` consumes `pad_mode`. Since `pad_mode` has been removed from `kwargs`, these tests are not equivalent to the ones in torchaudio, see [here](https://github.com/pytorch/audio/blob/98fe8b46af5582f602f58ca9f9633393f6a931a7/test/test_functional.py#L90).

Good catch. Will fix.